core/domain/feedback_jobs_one_off.py

```diff
@@ -100,3 +103,63 @@ def map(item):
     @staticmethod
     def reduce(key, value):
         pass
+
+
+class SuggestionMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for converting all suggestions from the old model to the
+    new model.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [feedback_models.SuggestionModel]
+
+    @staticmethod
+    def map(suggestion):
+        suggestion_id = (
+            suggestion_models.TARGET_TYPE_EXPLORATION + '.' + suggestion.id)
+        thread = feedback_models.FeedbackThreadModel.get_by_id(suggestion.id)
+        if thread.status == feedback_models.STATUS_CHOICES_OPEN:
+            status = suggestion_models.STATUS_RECEIVED
+        elif thread.status == feedback_models.STATUS_CHOICES_FIXED:
+            status = suggestion_models.STATUS_ACCEPTED
+        elif thread.status == feedback_models.STATUS_CHOICES_IGNORED:
+            status = suggestion_models.STATUS_REJECTED
+
+        score_category = (
+            suggestion_models.SCORE_TYPE_CONTENT +
+            suggestion_models.SCORE_CATEGORY_DELIMITER +
+            exp_services.get_exploration_by_id(
+                suggestion.exploration_id).category)
+
+        # For old accepted/rejected suggestions, the audio_translations field
+        # below value is not derivable as it is set only when the
+        # suggestion is accepted and that exploration version is unknown to us.
+        # We will never need to access the change_cmd parameter for these
+        # suggestions. For suggestions still in review, when the suggestion is
+        # accepted the audio_translations field will be updated. So while
+        # migrating we can set it to {} without losing required data.
+        change_cmd = {
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+            'state_name': suggestion.state_name,
+            'new_value': {
+                'html': suggestion.get_suggestion_html(),
+                'audio_translations': {}
+            }
+        }
+
+        suggestion_models.GeneralSuggestionModel(
+            id=suggestion_id,
+            suggestion_type=suggestion_models.SUGGESTION_EDIT_STATE_CONTENT,
```

Comment:
Just adding a note here. The name of the constant was changed from SUGGESTION_EDIT_STATE_CONTENT to SUGGESTION_TYPE_EDIT_STATE_CONTENT in PR #4982. Will edit this here if that PR goes in before this. Else will edit it in the other PR.

Reply:
4982 has been merged :)

Reply:
Done